clicache/integration-test/ThinClientRegionQueryTests.cs

```diff
@@ -182,7 +182,7 @@ public void StepThreeRQ()
 
         ISelectResults<object> results = region.Query<object>(qrystr.Query);
 
-        if (results.Size != QueryStatics.RegionQueryRowCounts[qryIdx])
+        if (results.Size != (ulong) QueryStatics.RegionQueryRowCounts[qryIdx])
```

Comment:
Prefer 'as' keyword to c style cast in c#.

Reply:
Keeping consistent with other casts in C# sources. Also, my understanding of `as` is it would result in an auto boxing of value type returned by call and then be unboxed to compare with the value type of `result.Size`. Since `QueryStatics.RegionQueryRowCounts[]` contains non-nullable values using the `as` keyword is less desirable.

Reply:
`as type` is to `dynamic_cast<type>` as `(type)` is to `static_cast<type>`... ish.

Reply:
The real advantage of `as` over c-style cast is that a c-style cast throws an InvalidCastException while `as` will result in a null value which can then be checked gracefully.  It is nicer to see a test failure due to an ASSERT_NOT_NULL (not sure thats the actual assert method or not) than an InvalidCastException.

```
var resultsSize = QueryStatics.RegionQueryRowCounts[qryIdx] as ulong;

ASSERT_NOT_NULL(resultsSize, "Results size was not a ulong");
....
```

Reply:
The `resultSize` is the expected value not the actual value under test. The type just needs to be nudged to fix differences between APIs in C++ / C#.

Reply:
Subtle difference - the next line would be `if (results.Size != resultsSize)` per my example above, but again probably doesn't buy us much here.